File: core/src/main/java/com/xuxue/spider/core/pluging/SpiderEvent.java
package com.xuxue.spider.core.pluging;

import com.xuxue.spider.core.cache.ResultIteamCache;
import com.xuxue.spider.core.cache.TaskCache;
import com.xuxue.spider.core.task.Task;

/**
 * Created by HanHan on 2016/6/15.
 */
public class SpiderEvent {

    private ResultIteamCache resultIteamCache;

    private TaskCache taskCache;

    private Object result;

    private ProcessStatus status;

    private Task task;

    private int responseCode;

    private Throwable e;

    public ProcessStatus getProcessStatus() {
        return this.status;
    }

    public void setProcessStatus(ProcessStatus status) {
        this.status = status;
    }

    public <T> T getResult() {
        return (T) result;
    }

    public Task getTask() {
        return this.task;
    }

    public void setTask(Task task) {
        this.task = task;
    }

    public <T> void setResult(T result) {
        this.result = result;
    }

    public ResultIteamCache getResultIteamCache() {
        return this.resultIteamCache;
    }

    public void setResultIteamCache(ResultIteamCache resultIteamCache) {
        this.resultIteamCache = resultIteamCache;
    }

    public TaskCache getTaskCache() {
        return this.taskCache;
    }

    public void setTaskCache(TaskCache taskCache) {
        this.taskCache = taskCache;
    }

    public int getResponseCode(){
        return this.responseCode;
    }

    public void setResponseCode(int responseCode){
        this.responseCode=responseCode;
    }

    public void setException(Throwable e){
        this.e=e;
    }

    public Throwable getException(){
        return this.e;
    }

}

File: core/src/main/scala/com/xuxue/spider/core/scala/event/SpiderEvent.scala
package com.xuxue.spider.core.scala.event

import com.xuxue.spider.core.scala.cache.Cache
import com.xuxue.spider.core.scala.result.ResultIteam
import com.xuxue.spider.core.scala.task.Task

/**
  * Created by HanHan on 2016/7/8.
  */
class SpiderEvent[T](val resultCache:Cache[ResultIteam],
                     val taskCache:Cache[Task],
                     val t:Either[T,Throwable],
                     val task:Task) {

}

object SpiderEvent{

}

class SpiderEventBuilder[T]{
  var t:Either[T,Throwable]= _
  var resultCache:Cache[ResultIteam]=_
  var task:Task=_
  var taskCache:Cache[Task]=_

  def setElement(either: Either[T,Throwable]):SpiderEventBuilder[T]={
    this.t=either
    this
  }

  def setResultCache(cache: Cache[ResultIteam]):SpiderEventBuilder[T]={
    this.resultCache=cache
    this
  }

  def setTask(task:Task):SpiderEventBuilder[T]={
    this.task=task
    this
  }

  def setTaskCache(taskCache:Cache[Task]):SpiderEventBuilder[T]={
    this.taskCache=taskCache
    this
  }

}